apps/arena/lib/arena/game/player.ex

```diff
@@ -362,4 +357,23 @@ defmodule Arena.Game.Player do
         player
     end
   end
+
+  ## Yes, we are pattern matching on exactly one mechanic. As of time writing we only have one mechanic per skill
+  ## so to simplify my life an executive decision was made to take thas as a fact
+  ## When the time comes to have more than one mechanic per skill this function will need to be refactored, good thing
+  ## is that it will crash here so not something we can ignore
+  defp calculate_duration(%{mechanics: [%{leap: leap}]}, position, direction) do
```

Comment:
This is not working. It should be:
`defp calculate_duration(%{mechanics: [leap: leap]}, position, direction) do`